### PR TITLE
Add method to update codescanning alert

### PR DESCRIPTION
**Pull request by ganeshkumarsv**
Adds method to update codescanning alert.
https://docs.github.com/en/rest/code-scanning?apiVersion=2022-11-28#update-a-code-scanning-alert

### CODE REVIEW

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:176]:
```suggestion
// CodeScanningAlertState specifies the state of a code scanning alert.
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:182]:
```suggestion
	// State sets the state of the code scanning alert and is a required field.
	// You must provide DismissedReason when you set the state to "dismissed".
	// It can be one of: "open", "dismissed".
	State string `json:"state"`
	// DismissedReason represents he reason for dismissing or closing the alert.
	// It is required when the state is "dismissed".
	// It can be one of: "false positive", "won't fix", "used in tests".
	DismissedReason *string `json:"dismissed_reason,omitempty"`
	// DismissedComment is associated with the dismissal of the alert.
	DismissedComment *string `json:"dismissed_comment,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @ganeshkumarsv !

**Review by gmlewis — CHANGES_REQUESTED** [review]
Can you please run `gofmt` on the newly-changed files and also make sure to run "go generate ./..." and push the changed files to this PR to get all the tests to pass?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Oh, and of course, please run "go test ./..." to catch the error in the test file.

**Comment by ganeshkumarsv** [comment]
Thanks for the commit suggestion @gmlewis I fixed the  test cases and generators.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:181]:
```suggestion
	// You must also provide DismissedReason when you set the state to "dismissed".
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:182]:
```suggestion
	// State can be one of: "open", "dismissed".
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:184]:
```suggestion
	// DismissedReason represents the reason for dismissing or closing the alert.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Sorry, @ganeshkumarsv - it looks like I made a couple mistakes in my suggestions.
Just a few minor tweaks, please.

**Review by gmlewis — APPROVED** [review]
Thank you, @ganeshkumarsv !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by AbbanMustafa — APPROVED** [review]

**Comment by ganeshkumarsv** [comment]
@gmlewis we got 2 approvals on the PR. can you please merge the PR and create a new release?

**Comment by gmlewis** [comment]
> @gmlewis we got 2 approvals on the PR. can you please merge the PR and create a new release?

Yes, but as for the new release, we already have breaking API changes and I would like to get https://github.com/google/go-github/pull/2646 approved and committed too before the major version change, so if you feel like reviewing that one, @ganeshkumarsv , that would speed the process along.  Thanks.

**Comment by gmlewis** [comment]
Thank you, @AbbanMustafa !
Merging.

**Comment by ganeshkumarsv** [comment]
Thanks @gmlewis. I approved the [PR](https://github.com/google/go-github/pull/2646)

**Comment by gmlewis** [comment]
@ganeshkumarsv - this is now available in the v50.0.0 release: https://github.com/google/go-github/releases/tag/v50.0.0

**Pull request merged**
